src/book/02-system/04-hardware/04-dynamixel-protocol.mdx

```diff
@@ -0,0 +1,388 @@
+
+| Line        | Logic-level | Description                                                                 | Optional |
+| ----------- | ----------- | --------------------------------------------------------------------------- | -------- |
+| Idle        | 1           | is where there is no data.                                                  | no       |
+| Start-bit   | 0           | is the beginning of the data.                                               | no       |
+| Data-bits   | ?           | are the bits of data from the least significant to the most significant.    | no       |
+| Parity-bits | ?           | tell whether there has been any corruption of the data during transmission. | yes      |
+| Stop-bit    | 1           | is the end of the data.                                                     | yes      |
+
+The receiver looks for the start-bit's falling edge. When it has found the edge, it begins to time and to sample each bit thereforth based on the agreed upon bitrate. Most setups are written by three symbols, e.g. 8N1 means that there are eight data-bits, no parity bits, and one stop-bit. Both Dynamixel protocols and many other serial ones are 8N1. Henceforth in this guide, this is the assumed framing.
+
+<Alert type="warn">
+
+Two UARTs must agree on the baudrate, framing (e.g. 8N1), and the signal-levels in order to communicate.
+
+</Alert>
+
+However, pure UART often only refers to abstract bits or logic levels, not voltages. Signalling techniques such as RS485, RS232, and TTL are what are further needed for a complete system to define the electrical behaviour as well as maybe flow-control (e.g. RS232).
+
+## Duplexity
+
+Firstly, one must understand the concept of duplexity since the half duplexity of the Dynamixel protocol is not always intuitive for beginners. There are three kinds of digital communication, namely simplex, full duplex, and half duplex.
+
+Simplexity is when data goes in only one direction, from Device-A to Device-B, where only the former transmits and only the latter receives. There is only one channel, and there may often be only one electrical signal. An analogy would be Alice giving instructions to Bob through a loud-speaker very far away.
+
+![In simplexity, data always flows in only one direction.](./images/dynamixel_protocol/simplexity.drawio.png 'In simplexity, data always flows in only one direction.')
+
+Full duplexity is when data can go in both directions at the same time, between Device-A and Device-B. There are two channels, one for each direction such that one could think of each channel as simplex. There is no clash when both Device-A and Device-B send data at the same time since the two channels are separate. An analogy would be Alice and Bob writing letters or texting.
```

Comment:
```suggestion
Full duplexity is when data can go in both directions at the same time, between Device-A and Device-B. There are two channels, one for each direction. This is like having two simplex channels. There is no clash when both Device-A and Device-B send data at the same time since the two channels are separate. An analogy would be Alice and Bob writing letters or texting.
```